packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-header/index.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { css, spacing, Body, Icon } from '@mongodb-js/compass-components';
+import { connect } from 'react-redux';
+import type { ConnectedProps } from 'react-redux';
+import type { Dispatch } from 'redux';
+
+import PipelineStages from './pipeline-stages';
+import PipelineActions from './pipeline-actions';
+import {
+  getSavedPipelines,
+  savedPipelinesListToggle,
+} from '../../../modules/saved-pipeline';
+
+const containerStyles = css({
+  display: 'grid',
+  gap: spacing[4],
+  gridTemplateAreas: '"pipelineTextAndOpen pipelineStages pipelineAction"',
+  gridTemplateColumns: 'min-content',
+  alignItems: 'center',
+});
+
+const pipelineTextAndOpenStyles = css({
+  display: 'flex',
+  gridArea: 'pipelineTextAndOpen',
+  gap: spacing[2],
+});
+const openSavedPipelinesStyles = css({
+  border: 'none',
+  backgroundColor: 'transparent',
+  lineHeight: 1,
+  marginTop: spacing[1],
+  display: 'flex',
+});
+
+const pipelineStagesStyles = css({
+  gridArea: 'pipelineStages',
+});
+const pipelineActionStyles = css({
+  gridArea: 'pipelineAction',
+  justifySelf: 'end',
+});
+
+const PipelineHeader: React.FunctionComponent<PipelineHeaderProps> = ({
+  onShowSavedPipelines,
+}) => {
+  return (
+    <div className={containerStyles} data-testid="pipeline-header">
+      <div className={pipelineTextAndOpenStyles}>
+        <Body weight="medium">Pipeline</Body>
+        <button
+          data-testid="pipeline-toolbar-open-pipelines-button"
+          onClick={() => onShowSavedPipelines()}
+          className={openSavedPipelinesStyles}
+          aria-label="Open saved pipelines"
+        >
+          <Icon glyph="Folder" />
+          <Icon glyph="CaretDown" />
+        </button>
+      </div>
+      <div className={pipelineStagesStyles}>
+        <PipelineStages />
+      </div>
+      <div className={pipelineActionStyles}>
+        <PipelineActions />
+      </div>
+    </div>
+  );
+};
+
+const mapDispatch = (dispatch: Dispatch) => ({
+  onShowSavedPipelines: () => {
+    // todo: fix dispatch
```

Comment:
Leaving this as is here - this is the old implementation and I don't want to fix this in this PR (as it has already exploded the limits). Fixing this will require changing it at many places.

Reply:
You don't need to refactor whole stores, but at least adding a new thunk action export that keeps track of all the logic required to open a saved pipelines list from saved pipeline actions that combines the two instead of directly using them here would be better. That way when we eventually refactoring this code away we will at least not need to hunt down every reference to these setter actions everywhere and will just have one import here that already matches what we want it to do.